scrapy/selector/unified.py

```diff
@@ -47,113 +26,34 @@ def _response_from_text(text, st):
               body=to_bytes(text, 'utf-8'))
 
 
-class Selector(object_ref):
-
-    __slots__ = ['response', 'text', 'namespaces', 'type', '_expr', '_root',
-                 '__weakref__', '_parser', '_csstranslator', '_tostring_method']
+class Selector(ParselSelector, object_ref):
 
-    _default_type = None
-    _default_namespaces = {
-        "re": "http://exslt.org/regular-expressions",
+    __slots__ = ['response']
 
-        # supported in libxslt:
-        # set:difference
-        # set:has-same-node
-        # set:intersection
-        # set:leading
-        # set:trailing
-        "set": "http://exslt.org/sets"
-    }
-    _lxml_smart_strings = False
+    def __init__(self, response=None, text=None, type=None, root=None, _root=None, **kwargs):
+        st = _st(response, type or self._default_type)
 
-    def __init__(self, response=None, text=None, type=None, namespaces=None,
-                 _root=None, _expr=None):
-        self.type = st = _st(response, type or self._default_type)
-        self._parser = _ctgroup[st]['_parser']
-        self._csstranslator = _ctgroup[st]['_csstranslator']
-        self._tostring_method = _ctgroup[st]['_tostring_method']
+        if root is None and _root is not None:
```

Comment:
Okay, so this is the only bit that have me a tiny bit anxious.

Is it okay to ignore `_root` when we get a `root` argument (assuming `root` precedence will speak for itself), or should we always warn whenever there is a value for `_root`?